src/target/cortexm.c

```diff
@@ -1122,3 +1124,61 @@ static int cortexm_hostio_request(target *t)
 
 	return t->tc->interrupted;
 }
+
+/**
+ * This command relies on data in the flash structure linked list.
+ * With one argument, it erases the page that contains the specified address,
+ * a second argument can be used to specify guaranteed length erased from start
+ * It checks for right number of arguments, correct number format and correct
+ * range.
+ **/
+bool cortexm_cmd_erase_range(target *t, int argc, char *argv[])
+{
+	if ((argc != 2) && (argc != 3))
+		{
+			tc_printf(t, "usage: monitor erase_range <start> [<end>]\n");
+			return false;
+		}
+
+	char *eos;
+	uint32_t addr = strtoul(argv[1], &eos, 0);
+	if (*eos)
```

Comment:
Looking at the docs for [`strtoul`](https://www.tutorialspoint.com/c_standard_library/c_function_strtoul.htm), looks like checking `addr != 0` should be the error check. Dereferencing `eos` just makes me paranoid.

Reply:
Do you think 
	errno = 0;
	uint32_t addr = strtoul(argv[1], NULL, 0);
	if ((addr == 0) && (errno))
is the right thing?

Reply:
Personally I think
```c
if ((addr == 0) || (errno))
```
makes more sense and is a bit more safe since either or is an error condition, but I think what you've done should work too